extensions/amp-analytics/0.1/transport.js

```diff
@@ -19,42 +19,68 @@ import {
   parseUrl,
   checkCorsUrl,
 } from '../../../src/url';
+import {urls} from '../../../src/config';
+import {createElementWithAttributes} from '../../../src/dom';
 import {dev, user} from '../../../src/log';
+import {getMode} from '../../../src/mode';
 import {loadPromise} from '../../../src/event-helper';
 import {timerFor} from '../../../src/services';
 import {removeElement} from '../../../src/dom';
-import {setStyle} from '../../../src/style';
+import {setStyle, setStyles} from '../../../src/style';
+import {hasOwn} from '../../../src/utils/object';
+import {IframeMessagingClient} from '../../../3p/iframe-messaging-client';
+import {MessageTypes} from '../../../src/3p-analytics-common';
 
-/** @const {string} */
+/** @private @const {string} */
 const TAG_ = 'amp-analytics.Transport';
 
-/**
- * @param {!Window} win
- * @param {string} request
- * @param {!Object<string, string>} transportOptions
- */
-export function sendRequest(win, request, transportOptions) {
-  assertHttpsUrl(request, 'amp-analytics request');
-  checkCorsUrl(request);
-  if (transportOptions['beacon'] &&
-      Transport.sendRequestUsingBeacon(win, request)) {
-    return;
-  }
-  if (transportOptions['xhrpost'] &&
-      Transport.sendRequestUsingXhr(win, request)) {
-    return;
-  }
-  if (transportOptions['image']) {
-    Transport.sendRequestUsingImage(win, request);
-    return;
-  }
-  user().warn(TAG_, 'Failed to send request', request, transportOptions);
-}
+/** @private @const {number} */
+const MESSAGE_THROTTLE_TIME_ = 100;
+
+/** @private @const {number} */
+const MAX_QUEUE_SIZE_ = 100;
 
 /**
  * @visibleForTesting
  */
 export class Transport {
+  /**
+   * @param {!string} type The value of the amp-analytics tag's type attribute
+   */
+  constructor(type) {
+    /** @private @const {string} */
+    this.id_ = Transport.createUniqueId_();
```

Comment:
Easy to guess then. I suppose an attacker would still need the window handle of the other xframe they're attacking. But didn't want to make it easy for them.